code/game/objects/structures/doors/_door.dm

```diff
@@ -0,0 +1,195 @@
+/obj/structure/door
+	name = "door"
+	icon = 'icons/obj/doors/material_doors.dmi'
+	icon_state = "metal"
+	hitsound = 'sound/weapons/genhit.ogg'
+	material_alteration = MAT_FLAG_ALTERATION_NAME | MAT_FLAG_ALTERATION_COLOR | MAT_FLAG_ALTERATION_COLOR
+	maxhealth = 50
+	density =  TRUE
+	anchored = TRUE
+	opacity =  TRUE
+
+	var/changing_state = FALSE
+	var/icon_base
+	var/datum/lock/lock
+
+/obj/structure/door/Initialize()
+	. = ..()
+	if(!istype(material))
+		return INITIALIZE_HINT_QDEL
+	if(lock)
+		lock = new /datum/lock(src, lock)
+	if(!icon_base)
+		icon_base = material.door_icon_base
+	update_nearby_tiles(need_rebuild=TRUE)
```

Comment:
Sometimes for this type of obj, it's useful to call open() or close() or similar to make sure that the state it's in is actually a valid state, unless all permutations of the state vars are valid. They are sometimes set inconsistently in the map.

Reply:
The only state var it has is `density` unless you include the 'do not mess with this door' flag. If someone is setting it on the map that should really fail to work so they go and fix it.